test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedDeserializerUndeclaredTests.cs

```diff
@@ -1029,6 +1036,45 @@ public void ReadOpenEntryUndeclaredEmptyComplexPropertiesTest()
                 .RawValue);
         }
 
+        [Fact]
+        public void ReadOpenEntryUndeclaredPrimitiveCollectionPropertiesWithoutODataTypeAsODataCollectionTest()

```

Comment:
What's the difference between this and the `ReadOpenEntryUndeclaredCollectionPropertiesWithoutODataTypeAsODataCollectionTest()` test here: https://github.com/OData/odata.net/pull/2547/files#diff-6df71d1b0e678332630895edae51af726cdc219427a6d163c2cd226339eccccdR844 ?

Reply:
Perhaps you could add a test that verifies untyped collections of complex objects are also supported?

Reply:
I'm specifically trying to avoid adding this scope to the change. I agree that we should support it, and if this change *does* support it, that's great. I have a list of 9 items (this concern included) that need to be worked on after this that I plan to talk about in the next triage meeting, but I am hoping to not support all of them as part of this PR. 

Does that sound alright to you?

Reply:
To answer your other question, the existing test treats the dynamic property as untyped, while this test treats it as a collection of untyped values. This is because the existing test does not enable the new behavior flag in `ODataMessageReaderSettings`.

Reply:
Thanks for the clarifications, that sounds alright.